packages/blocks/src/store/selectors.js

```diff
@@ -95,6 +106,44 @@ export function getBlockVariations( state, blockName, scope ) {
 	} );
 }
 
+// TODO jsdoc and experimental and tests
+export const getBlockTypeWithVariationInfo = createRegistrySelector(
```

Comment:
I think so too, but mostly because of the `label`. It probably should be better to handle label where needed, but other than that I don't believe using two stores are a lot..

Before combining them I had a selector in `block` store similar to `getBlockType` but also checking for variations and each of the components (BlockIcon etc..) would retrieve the other info (attributes) from the `block-editor` store. The exact same code..